### PR TITLE
Tag Helm Charts with tagged versions

### LINKED ISSUE #746: Use tagged version for Helm Chart version
Charts published by CI are using `git describe` format for Chart Version. Instead, for tagged versions we should use tag name the same we do for container images.

### PR DESCRIPTION
**Pull request by zimnx**
Fixes #746

### CODE REVIEW

**Comment by zimnx** [comment]
Tested on fork:
- tagged build sets HELM_CHART_VERSION variable to tagged version and hence override variable in Makefile. https://github.com/zimnx/scylla-operator/runs/3256214153?check_suite_focus=true 
- builds on master and feature branches doesn't set HELM_CHART_VERSION at all, so `GIT_TAG` is used.
https://github.com/zimnx/scylla-operator/runs/3256213562?check_suite_focus=true

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -259,9 +258,18 @@ jobs:
...
+
         source ./hack/lib/tag-from-gh-ref.sh
+        source ./hack/lib/semver.sh
+
         IMAGE_TAG=$( tag_from_gh_ref "${GITHUB_REF}" )
         echo "IMAGE_TAG=${IMAGE_TAG}" | tee -a ${GITHUB_ENV}
+
+        # Helm Chart version must be in semantic format.
+        if is_semver "${IMAGE_TAG}" ; then
+          echo "HELM_CHART_VERSION=${IMAGE_TAG}" | tee -a ${GITHUB_ENV}
```
Comment (tnozicka) [inline .github/workflows/go.yaml:271]: would it be enough just to use `git describe` without the `--long` option in the makefile?
https://github.com/scylladb/scylla-operator/blob/6439ee3ea61161855d4cf39da786238da5425495/Makefile#L56

Comment (zimnx) [reply .github/workflows/go.yaml]: yeah, good idea. Changed

**Review by tnozicka — APPROVED** [review]
/lgtm
thanks

**Pull request merged**